lib/timers.js

```diff
@@ -455,6 +480,9 @@ function Timeout(after) {
   this._idleStart = null;
   this._onTimeout = null;
   this._repeat = null;
+  this._asyncId = async_hooks.newUid();
+  if (async_hook_fields[kInit])
+    async_hooks.emitInit(this._asyncId, this, 'Timeout');
```

Comment:
Note: some sort of ID retention needs to be done for intervals...

(And maybe even timers that are re-inserted...)